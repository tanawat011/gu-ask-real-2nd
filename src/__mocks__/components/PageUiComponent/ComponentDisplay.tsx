// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck

export const ComponentDisplay = ({ code, component }) => {
  return (
    <div data-testid='component-display'>
      {component}
      {code}
    </div>
  )
}
