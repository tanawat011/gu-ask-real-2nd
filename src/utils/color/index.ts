export * from './hex2Rgb'
export * from './rgb2Hex'
