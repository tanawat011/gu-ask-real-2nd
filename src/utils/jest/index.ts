export * from './renderWithProviders'
export * from './twColor'
