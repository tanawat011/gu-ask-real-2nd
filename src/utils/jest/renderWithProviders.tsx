import type { RenderResult } from '@testing-library/react'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { RecoilRoot } from 'recoil'

const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  const queryClient = new QueryClient()

  return (
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <BrowserRouter>{children}</BrowserRouter>
      </RecoilRoot>
    </QueryClientProvider>
  )
}

interface RenderWithProvidersOptions {
  router?: {
    route: string
  }
  recoil?: object
}

export const renderWithProviders = (
  ui: JSX.Element,
  option?: RenderWithProvidersOptions,
): RenderResult => {
  const routerOption = { ...(option?.router || { route: '/' }) }
  window.history.pushState({}, 'Test page', routerOption.route)

  return {
    ...render(ui, { wrapper: AllTheProviders, ...routerOption }),
  }
}
