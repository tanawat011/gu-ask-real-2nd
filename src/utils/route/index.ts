export * from './menuGenerator'
export * from './renderRoutes'
export * from './routeGenerator'
