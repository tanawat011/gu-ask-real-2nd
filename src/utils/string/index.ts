export * from './camelToKebab'
export * from './camelToSentence'
export * from './replaceAll'
