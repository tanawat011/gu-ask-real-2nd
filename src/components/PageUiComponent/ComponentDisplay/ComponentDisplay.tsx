import tw, { styled } from 'twin.macro'

import { CodeDisplay } from 'components/PageUiComponent'
import { BORDER_THEME } from 'constants/twTheme'

const TwContainer = styled.div(() => [BORDER_THEME.BASE, tw`rounded-lg border`])
const TwWrapComponent = tw.div`grid gap-2 p-5`

type ComponentDisplayProps = {
  code: string
  components: React.ReactNode
  callback?: () => void
}

export const ComponentDisplay: React.FC<ComponentDisplayProps> = ({
  code,
  components,
  callback,
}) => {
  return (
    <TwContainer>
      <TwWrapComponent>{components}</TwWrapComponent>

      <CodeDisplay code={code} callback={callback} />
    </TwContainer>
  )
}
