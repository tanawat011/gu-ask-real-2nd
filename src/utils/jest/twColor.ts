import { hex2Rgb } from 'utils/color'

export const twColor = (hexColor: string, type?: 'bg' | 'border' | 'text') => {
  if (hexColor === 'transparent') {
    return 'transparent'
  }

  if (hexColor.length > 7) {
    return 'undefined'
  }

  const rgb = hex2Rgb(hexColor, { isArray: true }) as string[]

  if (!type) {
    type = 'bg'
  }

  return `rgb(${rgb.join(' ')} / var(--tw-${type}-opacity))`
}
