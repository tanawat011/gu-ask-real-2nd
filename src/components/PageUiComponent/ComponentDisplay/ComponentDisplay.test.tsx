import { renderWithProviders } from 'utils/jest'

import { ComponentDisplay } from './ComponentDisplay'

jest.mock('../CodeDisplay', () => ({
  CodeDisplay: () => <div>CodeDisplay</div>,
}))

describe('<ComponentDisplay />', () => {
  test('renders correctly', () => {
    expect.assertions(1)

    const { asFragment } = renderWithProviders(
      <ComponentDisplay code='' components={<div>xxx</div>} />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
