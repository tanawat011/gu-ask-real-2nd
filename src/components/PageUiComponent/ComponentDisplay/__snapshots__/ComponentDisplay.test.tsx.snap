// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentDisplay /> renders correctly 1`] = `
<DocumentFragment>
  .c1 {
  --tw-border-opacity: 1;
  border-color: rgb(229 231 235 / var(--tw-border-opacity));
  border-radius: 0.5rem;
  border-width: 1px;
}

.dark .c0 {
  --tw-border-opacity: 1;
  border-color: rgb(75 85 99 / var(--tw-border-opacity));
}

.c2 {
  display: grid;
  gap: 0.5rem;
  padding: 1.25rem;
}

<div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <div>
        xxx
      </div>
    </div>
    <div
      data-testid="code-display"
    >
      code
    </div>
  </div>
</DocumentFragment>
`;
